firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3819,18 +3819,20 @@ extension BrowserViewController: TabManagerDelegate {
 
         scrollController.tab = selectedTab
 
-        if !selectedTab.isFxHomeTab,
```

Comment:
cc @OrlaM This line was added in https://github.com/mozilla-mobile/firefox-ios/commit/d75728073f0fe31c9698298eb47bb2eea2513373, perhaps for FXIOS-10155? I don't have full context on the change but I retested 10155 with the changes here and couldn't replicate any obvious bugs. But if you have any other information on what this was originally added to fix please LMK.

Reply:
I'm not 100% clear on what you mean; race condition implies that there are two or more threads at play, but this is UI related code that should only execute on the main thread. Do you mean there is an order-of-operations problem that was happening previously? Do you remember the steps or details?

> Can you try wrap just that part of the code block in if isFxHomeTab

I did try that and unfortunately it causes the bug to return (audio/video leak).

Reply:
Yes it's an order of operations problem, I don't recall the repo steps. If it's working now I'm fine with merging it as is, I would like to see a follow up ticket to revisit why the web view would need to be updated for a home page tho as it suggests something off in how things are wired together. The fix is likely a bit of a refactor and a lot more risk tho so I'm fine with not doing it now.

Reply:
Filed FXIOS-10612